From 674f835e59e5abfac5a288bfe9686bfd54eb948f Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Alfonso=20S=C3=A1nchez-Beato?=
 <alfonso.sanchez-beato@canonical.com>
Date: Tue, 21 Sep 2021 09:29:19 +0200
Subject: [PATCH 1/2] main: add ignore-serial-consoles option

Add --ignore-serial-consoles option so we can ignore serial consoles
also when starting plymouthd and not only from the kernel command
line.

Upstream MP: https://gitlab.freedesktop.org/plymouth/plymouth/-/merge_requests/143
---
 src/main.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/src/main.c b/src/main.c
index 03adc1f..e018b7f 100644
--- a/src/main.c
+++ b/src/main.c
@@ -2089,6 +2089,7 @@ main (int    argc,
         bool no_boot_log = false;
         bool no_daemon = false;
         bool debug = false;
+        bool ignore_serial_consoles = false;
         bool attach_to_session;
         ply_daemon_handle_t *daemon_handle = NULL;
         char *mode_string = NULL;
@@ -2112,6 +2113,7 @@ main (int    argc,
                                         "kernel-command-line", "Fake kernel command line to use", PLY_COMMAND_OPTION_TYPE_STRING,
                                         "tty", "TTY to use instead of default", PLY_COMMAND_OPTION_TYPE_STRING,
                                         "no-boot-log", "Do not write boot log file", PLY_COMMAND_OPTION_TYPE_FLAG,
+                                        "ignore-serial-consoles", "Ignore serial consoles", PLY_COMMAND_OPTION_TYPE_FLAG,
                                         NULL);
 
         if (!ply_command_parser_parse_arguments (state.command_parser, state.loop, argv, argc)) {
@@ -2132,6 +2134,7 @@ main (int    argc,
                                         "no-boot-log", &no_boot_log,
                                         "no-daemon", &no_daemon,
                                         "debug", &debug,
+                                        "ignore-serial-consoles", &ignore_serial_consoles,
                                         "debug-file", &debug_buffer_path,
                                         "pid-file", &pid_file,
                                         "tty", &tty,
@@ -2269,7 +2272,8 @@ main (int    argc,
         if (isnan (state.device_timeout) || state.device_timeout <= 0.0)
                 state.device_timeout = 8.0;
 
-        if (ply_kernel_command_line_has_argument ("plymouth.ignore-serial-consoles"))
+        if (ply_kernel_command_line_has_argument ("plymouth.ignore-serial-consoles") ||
+            ignore_serial_consoles == true)
                 device_manager_flags |= PLY_DEVICE_MANAGER_FLAGS_IGNORE_SERIAL_CONSOLES;
 
         if (ply_kernel_command_line_has_argument ("plymouth.ignore-udev") ||
-- 
2.25.1

